### PR TITLE
Q7 align spectra

### PR DESCRIPTION
**Pull request by astrofle**
In this PR:
* Method to shift `Spectrum` given a number of channels.
* Refactor of the code used to shift. Now it can be found in `spectra.core`.
* Method to find the shift between two `Spectrum`.
* Method to align one `Spectrum` to another.
* Function to average spectra.
* Tests for all of the above.
* A notebook showing how to use the functions.

### CODE REVIEW

File: src/dysh/spectra/spectrum.py [hunk 1]
```diff
@@ -1400,3 +1509,66 @@ def spectrum_reader_gbtidl(fileobj, **kwargs):
     # registry.register_writer("ipac", Spectrum, ascii_spectrum_reader_ipac)
     # registry.register_writer("votable", Spectrum, spectrum_reader_votable)
     # registry.register_writer("mrt", Spectrum, spectrum_reader_mrt)
+
+
+def average_spectra(spectra, equal_weights=False):
```
Comment (mpound) [inline src/dysh/spectra/spectrum.py:1514]: This should be moved to core.py and eventually replace use of  average() in time averaging.

File: src/dysh/spectra/spectrum.py [hunk 2]
```diff
@@ -1400,3 +1509,66 @@ def spectrum_reader_gbtidl(fileobj, **kwargs):
...
+    """
+    Average `spectra`.
+
+    Parameters
```
Comment (mpound) [inline src/dysh/spectra/spectrum.py:1518]: Future enhancement: align=T/F to align all spectra to the first in the list.

File: src/dysh/spectra/spectrum.py (hunk 1 above)
Comment (astrofle) [reply src/dysh/spectra/spectrum.py:1514]: I couldn't figure out how to check for `Spectrum` instances from core due to a circular import issue.

Comment (astrofle) [reply src/dysh/spectra/spectrum.py:1514]: Also, `core.average` and `spectrum.average_spectra` serve two different use cases. The former is for averaging inside a `Scan`, the later is for averaging `Spectrum`. To replace one with the other we would need to turn every integration into a `Spectrum`, which would be slower than needed.

File: src/dysh/spectra/spectrum.py (hunk 2 above)
Comment (astrofle) [reply src/dysh/spectra/spectrum.py:1518]: Created issue #402

File: src/dysh/spectra/spectrum.py [hunk 3]
```diff
@@ -818,6 +929,14 @@ def fake_spectrum(cls, nchan=1024, **kwargs):
         nchan : int, optional
             Number of channels. The default is 1024.
 
+        seed : {None, int, array_like[ints], SeedSequence, BitGenerator, Generator}, optional
```
Comment (mpound) [inline src/dysh/spectra/spectrum.py:932]: These should be e.g., `~numpy.random.SeedSequence`  so the docs will link to the numpy docs/

**Review by mpound — APPROVED** [review]
Just fix that one doc issue in fake_spectra.  Very nice work!

**Pull request merged**